src/v0/destinations/mp/deleteUsers.js

```diff
@@ -68,6 +64,81 @@ const userDeletionHandler = async (userAttributes, config) => {
   };
 };
 
+const createDeletionTask = async (userAttributes, config) => {
+  const { token, gdprApiToken } = config;
+
+  if (!gdprApiToken) {
+    throw new ConfigurationError(
+      'GDPR Api Token is a required field for creating deletion task in mixpanel',
+    );
+  }
+
+  const endpoint = getCreateDeletionTaskEndpoint(token);
+  const headers = {
+    'Content-Type': JSON_MIME_TYPE,
+    Authorization: `Bearer ${gdprApiToken}`,
+  };
+
+  const groupedUserAttributes = _.groupBy(
+    userAttributes,
+    (attribute) => attribute.complianceType || DEFAULT_COMPLIANCE_TYPE,
+  );
+
+  // chunkedUserIds = [[u1,u2,u3,..batchSize],[u1,u2,u3,..batchSize]..]
+  // ref : https://developer.mixpanel.com/docs/privacy-security#create-a-deletion-task
+  const chunkedUserIds = _.mapValues(groupedUserAttributes, (group) =>
+    _.chunk(_.map(group, 'userId'), DISTINCT_ID_MAX_BATCH_SIZE),
+  );
+  await Promise.all(
+    Object.keys(chunkedUserIds).map(async (complianceType) => {
+      const batchEvents = chunkedUserIds[complianceType];
+      await Promise.all(
+        batchEvents.map(async (batchEvent) => {
+          const request = {
+            distinct_ids: batchEvent,
+            compliance_type: complianceType,
+          };
+          const deletionResponse = await httpPOST(endpoint, request, { headers });
+          const handledDelResponse = processAxiosResponse(deletionResponse);
```

Comment:
Can we use `handleHttpRequest` method instead of `httpPOST` ?

Reply:
updated